Examples/ObjectiveC/RainFallExample.m

```diff
@@ -0,0 +1,140 @@
+}
+
+- (CGRect)perfectSliderFrame {
+    if (UIApplication.sharedApplication.statusBarOrientation == UIDeviceOrientationPortrait) {
+        return CGRectMake(10,
+                          self.view.frame.size.height - 60 - self.bottomLayoutGuide.length,
+                          self.view.frame.size.width - 20,
+                          20);
+    }
+    return CGRectMake(self.topLayoutGuide.length,
+                      self.view.frame.size.height - 60 - self.bottomLayoutGuide.length,
+                      self.view.frame.size.width - self.topLayoutGuide.length - 20,
+                      20);
+}
+
+- (void)didChangeOrientation {
+    [UIView animateWithDuration:0.3 animations:^{
+        self.slider.frame = [self perfectSliderFrame];
+    }];
+}
+
+- (void)timerTick {
+    self.timeIndex = self.timeIndex + 1;
+    self.radarLayer.predicate = [NSPredicate predicateWithFormat:@"idx == %d", self.timeIndex];
+}
+
+- (void)slideValueChanged:(UISlider *)silder event:(UIEvent *)event {
+    switch ([event allTouches].anyObject.phase) {
+        case UITouchPhaseBegan:
+            [self.timer invalidate];
+            break;
+        case UITouchPhaseEnded:
+            self.timer = [NSTimer scheduledTimerWithTimeInterval:0.15 target:self selector:@selector(timerTick) userInfo:nil repeats:YES];
+            break;
+        default:
+            break;
+    }
+    self.radarLayer.predicate = [NSPredicate predicateWithFormat:@"idx == %d", self.timeIndex];
+}
+
+- (void)addMapView {
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(28.22894310302858, 102.45434471254349) zoomLevel:2 animated:NO];
+    self.mapView.delegate = self;
+    [self.view addSubview:self.mapView];
+}
+
+- (void)addSider {
+    self.slider = [[UISlider alloc] init];
+    self.slider.autoresizingMask = UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin;
+    self.slider.minimumValue = 0;
+    self.slider.maximumValue = 40;
+    self.slider.value = 0;
+    [self.slider addTarget:self action:@selector(slideValueChanged:event:) forControlEvents:UIControlEventValueChanged];
+    [self.view addSubview:self.slider];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    self.slider.frame = [self perfectSliderFrame];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self addMapView];
+    [self addSider];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didChangeOrientation) name:UIDeviceOrientationDidChangeNotification object:nil];
+}
+
+#pragma mark - mapview delegate
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    NSURL *url = [NSURL URLWithString:@"mapbox://shenhongissky.6vm8ssjm"];
```

Comment:
For consistency with our other examples, let's add this to the `mapbox://examples` account. Let me know if I can help!